torch_harmonics/distributed/distributed_convolution.py

```diff
@@ -158,11 +152,12 @@ def _precompute_distributed_convolution_tensor_s2(in_shape, out_shape, kernel_sh
         out_vals.append(vals)
 
     # concatenate the indices and values
-    out_idx = torch.cat(out_idx, dim=-1)
-    out_vals = torch.cat(out_vals, dim=-1)
+    out_idx = torch.cat(out_idx, dim=-1).to(torch.long).contiguous()
```

Comment:
why is the cast necessary here?

Reply:
It is safer since we pass it to type sensitive C++ code.